src/QuickBooks/Provider.php

```diff
@@ -2,25 +2,88 @@
 
 namespace SocialiteProviders\QuickBooks;
 
-use SocialiteProviders\Manager\OAuth1\AbstractProvider;
-use SocialiteProviders\Manager\OAuth1\User;
+use Laravel\Socialite\Two\ProviderInterface;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
 
-class Provider extends AbstractProvider
+class Provider extends AbstractProvider implements ProviderInterface
 {
     /**
      * Unique Provider Identifier.
      */
     const IDENTIFIER = 'QUICKBOOKS';
 
+    /**
+     * Sandbox endpoint for retrieving user info.
+     */
+    const USER_SANDBOX_ENDPOINT = 'https://sandbox-accounts.platform.intuit.com/v1/openid_connect/userinfo';
+
+    /**
+     * Production endpoint for retrieving user info.
+     */
+    const USER_PRODUCTION_ENDPOINT = 'https://accounts.platform.intuit.com/v1/openid_connect/userinfo';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['com.intuit.quickbooks.accounting openid profile email phone address'];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase('https://appcenter.intuit.com/connect/oauth2', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://oauth.platform.intuit.com/oauth2/v1/tokens/bearer';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $endpoint = self::USER_PRODUCTION_ENDPOINT;
+
+        if (env('QUICKBOOKS_ENV') === 'development') {
```

Comment:
can we add a config in the service's specific config please?
This will break if you run `artisan config:cache`

Reply:
for sure - do I need to make an additional PR for the README regarding this additional configuration too?